onnx/version_converter/IntraDomainConverter.h

```diff
@@ -0,0 +1,205 @@
+          std::string since = target_version.domain() + std::to_string(
+              current_opschemas.at(op)->since_version());
+          const auto& op_adapters = adapters.at(op_name);
+          if (op_adapters.find(since) != op_adapters.end() &&
+            op_adapters.at(since).find(target) != op_adapters
+            .at(since).end()) {
+            return *(op_adapters.at(since).at(target));
+          } else {
+            // TODO: Instead return NoUpwardsAdapter
+            throw "NoUpwardsAdapter";
+          }
+        }
+      } else {
+        // No adapters exist for the given op
+        // TODO: Instead return NoAdapterForOp
+        throw "NoAdapterForOp";
+      }
+    }
+
+    ONNX_NAMESPACE::ModelProto convert_version(
+        const ONNX_NAMESPACE::ModelProto& mp_in,
+        const OpSetID& initial_version,
+        const OpSetID& target_version) const {
+      ONNX_ASSERTM(strcmp(initial_version.domain().c_str(), target_version.domain()
+            .c_str()) == 0, "initial_version and target_version must have the same "
+          "domains");
+      for (auto it = mp_in.opset_import().begin(); it != mp_in.opset_import()
+          .end(); ++it) {
+        if (it->domain() == initial_version.domain()) {
+          ONNX_ASSERTM(initial_version.version() == it->version(),
+              "initial_version does not reflect current state of model");
+        }
+      }
+
+      std::shared_ptr<ONNX_NAMESPACE::Graph> g(ONNX_NAMESPACE::ImportModelProto(mp_in));
+      ONNX_ASSERTM(g.get() != nullptr,
+        "Warning: onnx version converter is unable to parse input model. "
+        "(The IR version of the ONNX model may be too old.)");
+
+      // TODO: Move to Inter-Domain Converter
+      // Get initial model versions
+      // std::vector<OpSetID> initial_versions = g->opset_versions();
+
+      // No conversion necessary if Model has single, equivalent opset version
+      // if (initial_versions.size() == 1 && initial_versions[0].version ==
+      //    target_version.version && initial_versions[0].domain ==
+      //    target_version.domain) {
+      //  return mp_in;
+      // }
+
+      // Check if target_version is valid
+      const std::unordered_map<std::string, std::pair<int, int>>& versions_map = OpSchemaRegistry::DomainToVersionRange::Instance().Map();
+      const std::string& search_domain = target_version.domain() == "ai.onnx" ? "" : target_version.domain();
+      const std::pair<int, int>& version_range = versions_map.at(search_domain);
+      ONNX_ASSERTM(target_version.version() >= version_range.first && target_version
+          .version() <= version_range.second,
+        "Warning: invalid target_version (must be between %s and %s",
+        version_range.first, version_range.second);
+      // Compile list of all ops used in the model
+      graph_node_list nodes = g->nodes();
+
+      const std::vector<OpSchema>& all_opschemas = ONNX_NAMESPACE::OpSchemaRegistry::get_all_schemas_with_history();
```

Comment:
probably you can remove all the ONNX_NAMESPACE in this file, because at the beginning, we already specify that we are using ONNX_NAMESPACE.